plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -780,11 +781,15 @@ public ConnectorTableProperties getTableProperties(ConnectorSession session, Con
                 }
             });
 
-            Iterable<FileScanTask> files = () -> lazyFiles.get().iterator();
+            Iterable<PartitionStruct> partitionStructs = () -> lazyFiles.get()
+                    .stream()
+                    .map(PartitionStruct::new)
+                    .distinct()
+                    .iterator();
```

Comment:
Is it possible to test this change in `TestIcebergFileOperations` or somewhere?

Reply:
added a test in `io.trino.plugin.iceberg.TestMetadataQueryOptimization`